pkgs/tools/system/lanzaboote/tool.nix

```diff
@@ -0,0 +1,102 @@
+{ systemd
+, makeWrapper
+, stdenv
+, binutils-unwrapped
+, sbsigntool
+, makeRustPlatform
+, rustPlatform
+, rustc
+, cargo
+, fetchFromGitHub
+, lanzaboote-stub
+, lib
+, mkStdenvNoLibs
+, pkgsBuildTarget
+, buildPackages
+, overrideCC
+, writeShellScriptBin
+}:
+let
+  uefiPlatform = lib.systems.elaborate "${stdenv.hostPlatform.qemuArch}-windows" // {
+    rustc.config = "${stdenv.hostPlatform.qemuArch}-unknown-uefi";
+    useLLVM = true;
+  };
+  uefiLinker = writeShellScriptBin "lld-link" ''
```

Comment:
This is the implementation of @amjoseph-nixpkgs's idea in #231960, cc @rrbutani FWIW.

It works because Rust will correctly infer from `lld-link` that it has to pass `-flavor link` and fortunately, it works because no "before arguments" gets added here.

Let me know if this is too fragile.